201-front-door-api-management/azuredeploy.json

```diff
@@ -0,0 +1,1048 @@
+          "Http",
+          "Https"
+        ],
+        "patternsToMatch": [
+          "/*"
+        ],
+        "customForwardingPath": "/",
+        "forwardingProtocol": "MatchRequest",
+        "cacheConfiguration": {
+          "queryParameterStripDirective": "StripNone",
+          "dynamicCompression": "Enabled"
+        }
+      },
+      "metadata": {
+        "description": "This parameter contains the name and properties of the routing rule."
+      }
+    },
+    "frontDoorHealthProbeSettings": {
+      "type": "object",
+      "defaultValue": {
+        "name": "ApimHealthProbeSettings",
+        "intervalInSeconds": 30,
+        "path": "/status-0123456789abcdef",
+        "protocol": "Https"
+      },
+      "metadata": {
+        "description": "This parameter contains the name and properties of the health probe settings."
+      }
+    },
+    "deployWaf": {
+      "type": "bool",
+      "defaultValue": true,
+      "metadata": {
+        "description": "Specifies whether to deploy a global WAF policy in Front Door."
+      }
+    },
+    "wafPolicyName": {
+      "type": "string",
+      "defaultValue": "FrontDoorApimSampleWaf",
+      "metadata": {
+        "description": "The name of the WAF policy"
+      }
+    },
+    "wafMode": {
+      "type": "string",
+      "allowedValues": [
+        "Detection",
+        "Prevention"
+      ],
+      "defaultValue": "Detection",
+      "metadata": {
+        "description": "Describes if it is in detection mode or prevention mode at policy level."
+      }
+    },
+    "allowTrafficOnlyFromFrontDoor": {
+      "type": "bool",
+      "defaultValue": false,
+      "metadata": {
+        "description": "Specifies whether to allow traffic to API Management public endpoints only from Front Door."
+      }
+    }
+  },
+  "variables": {
+    "nsgId": "[resourceId('Microsoft.Network/networkSecurityGroups', parameters('virtualNetwork').subnets[0].nsg)]",
+    "vnetId": "[resourceId('Microsoft.Network/virtualNetworks', parameters('virtualNetwork').name)]",
+    "apimId": "[resourceId('Microsoft.ApiManagement/service', parameters('apimServiceName'))]",
+    "apimHostname": "[concat(toLower(parameters('apimServiceName')), '.azure-api.net')]",
+    "appInsightsId": "[resourceId('Microsoft.Insights/components', parameters('appInsightsName'))]",
+    "apimSubnetId": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('virtualNetwork').name, parameters('virtualNetwork').subnets[0].name)]",
+    "workspaceId": "[resourceId('Microsoft.OperationalInsights/workspaces/', parameters('workspaceName'))]",
+    "loggerId": "[resourceId('Microsoft.ApiManagement/service/loggers', parameters('apimServiceName'), parameters('appInsightsName'))]",
+    "frontDoorId": "[resourceId('Microsoft.Network/frontdoors', parameters('frontDoorName'))]",
+    "frontDoorBackendPoolId": "[concat(variables('frontDoorId'), '/backendPools/', parameters('frontDoorBackendPool').name)]",
```

Comment:
remove concat and use resourceId for all resourceIds

Reply:
Solved. I replaced the previous resource Ids variables with the following:
"frontDoorId": "[resourceId('Microsoft.Network/frontdoors', parameters('frontDoorName'))]",
        "frontDoorBackendPoolId": "[resourceId('Microsoft.Network/frontdoors/backendPools', parameters('frontDoorName'), parameters('frontDoorBackendPool').name)]",
        "frontDoorHealthProbeSettingsId": "[resourceId('Microsoft.Network/frontdoors/healthProbeSettings', parameters('frontDoorName'), parameters('frontDoorHealthProbeSettings').name)]",
        "frontDoorLoadBalancerId": "[resourceId('Microsoft.Network/frontdoors/loadBalancingSettings', parameters('frontDoorName'), parameters('frontDoorBackendPool').loadBalancerName)]",
        "frontDoorFrontedEndpointId": "[resourceId('Microsoft.Network/frontdoors/frontendEndpoints', parameters('frontDoorName'), parameters('frontDoorFrontendEndpoint').name)]",
        "wafPolicyId": "[resourceId('Microsoft.Network/FrontDoorWebApplicationFirewallPolicies', parameters('wafPolicyName'))]",